sklearn/feature_selection/from_model.py

```diff
@@ -68,11 +71,16 @@ def transform(self, X, threshold=None):
 
         # Retrieve threshold
         if threshold is None:
-            if hasattr(self, "penalty") and self.penalty == "l1":
+            threshold = getattr(self, "threshold", None)
+
+        if threshold is None:
+            # Lasso has a l1 penalty but no penalty param.
+            if (hasattr(self, "penalty") and self.penalty == "l1" or
+                'Lasso' in self.__class__.__name__):
```

Comment:
Oh, I see work has already been done in deprecating _LearntSelectorMixin and having a SelectfromModel estimator. How far is it from being merged?